### ISSUE TITLE
[AGENT][S2] the pre-push gtest filter has no drift guard while two siblings do

### ISSUE THREAD

**Issue by kekzl** [kekzl/imp#1586]
## Fact
- The pre-push gate's 12-suite gtest filter is an unguarded string literal. The repo guards two sibling filters against exactly this failure mode; this one, which carries the only batch-invariance assertion in the tree, is not guarded.
- `scripts/verify.sh:282` — the unguarded filter
- `CMakeLists.txt:933-936` — `guard_e2e_lane_split`, guarding a different filter
- `CMakeLists.txt:945-948` — `guard_det_suite_filter`, guarding a third filter
- `scripts/verify.sh:278-281` — the comment stating what would be lost

## Evidence
```
# scripts/verify.sh:278-283 - the filter and why ForwardPassTest is in it
        # ForwardPassTest is here for the same reason: it is SKIP_IF_NO_CUDA, so
        # CI cannot run it, and DecodeLogitsInvariantToBatchComposition (#1314)
        # is the only assert in the tree that a sequence's logits do not depend
        # on its batch neighbours — the class #1044/#1045 came from.
        FILTER="TensorTest.*:GgufLoaderTest.*:Tokenizer*:ChatTemplate*:KVCache*:GemmTest.*:FP8GemmTest.*:SamplingTest.*:SoftmaxTest.*:AttentionTest.*:VramBudget*:ForwardPassTest.*"
        if "$TESTS_BIN" --gtest_filter="$FILTER" >/tmp/imp_verify_tests.log 2>&1; then
            pass "fast gtest filter"

# tests/CLAUDE.md:43-45 - the failure mode, already written down
- **`--gtest_filter` on a `TEST_P` or `TYPED_TEST` suite without wildcards
  matches zero tests and reports `PASSED`.** `DetEvalE2ETest.*` matches nothing;
  `*DetEvalE2ETest*` is correct. A CPU-lane guard now asserts this.

# CMakeLists.txt:933-936 and :945-948 - the two guards that exist
    add_test(NAME guard_e2e_lane_split
        COMMAND ${CMAKE_COMMAND} -E env
                ${CMAKE_CURRENT_SOURCE_DIR}/scripts/check_e2e_lane_split.sh
                $<TARGET_FILE:test-e2e> "${_unit_e2e_filter}")
    add_test(NAME guard_det_suite_filter
        COMMAND ${CMAKE_COMMAND} -E env
                ${CMAKE_CURRENT_SOURCE_DIR}/scripts/check_det_suite_filter.sh
                $<TARGET_FILE:test-e2e> "${_det_e2e_filter}")

# CMakeLists.txt:926-927 - the sync problem is known and handled by a comment only
    # NOTE: scripts/verify.sh carries a literal copy of this filter (container
    # path has no build/ dir) — keep both in sync when changing it.
```
No TEST_P/TYPED_TEST trap is live in the filter today, so the gate currently works:
```
$ for s in TensorTest GgufLoaderTest Tokenizer ChatTemplate KVCache GemmTest \
           FP8GemmTest SamplingTest SoftmaxTest AttentionTest VramBudget ForwardPassTest; do
    n=$(grep -rn "TEST_P($s\|TYPED_TEST($s\|TYPED_TEST_SUITE($s\|INSTANTIATE_TEST_SUITE_P($s" tests/ | wc -l)
    echo "$s: TEST_P/TYPED hits=$n"; done
TensorTest: TEST_P/TYPED hits=0        GgufLoaderTest: TEST_P/TYPED hits=0
Tokenizer: TEST_P/TYPED hits=0         ChatTemplate: TEST_P/TYPED hits=0
KVCache: TEST_P/TYPED hits=0           GemmTest: TEST_P/TYPED hits=0
FP8GemmTest: TEST_P/TYPED hits=0       SamplingTest: TEST_P/TYPED hits=0
SoftmaxTest: TEST_P/TYPED hits=0       AttentionTest: TEST_P/TYPED hits=0
VramBudget: TEST_P/TYPED hits=0        ForwardPassTest: TEST_P/TYPED hits=0
```

## Impact
- A rename or a suite move silently drops that suite from the pre-push gate: gtest matches zero tests for that entry, the other 11 still run, `verify.sh:283` sees exit 0 and prints `PASS fast gtest filter`. Nothing reports the loss.
- The suite most exposed is `ForwardPassTest`, whose own comment two lines above says it holds the only batch-invariance assertion in the tree, and which cannot be caught by CI because it is `SKIP_IF_NO_CUDA` and CI has no GPU runner (`.github/workflows/ci.yml:507`).
- Blast radius: the pre-push gate for all 12 suites, i.e. tensors, GGUF loading, tokenizer, chat templates, KV cache, GEMM, FP8 GEMM, sampling, softmax, attention, VRAM budget, forward pass.

## Repro
```bash
# NOT RUN - static, shows the guard coverage gap:
grep -n 'FILTER=' /home/kekz/github.com/kekzl/imp/scripts/verify.sh
grep -n 'add_test(NAME guard' /home/kekz/github.com/kekzl/imp/CMakeLists.txt
# NOT RUN - demonstrates the silent pass (needs the built binary):
./build/imp-tests --gtest_filter='NoSuchSuite.*'; echo "exit=$?"   # expect 0
```

## Fix direction
- Add a CPU-lane guard in the shape of `check_det_suite_filter.sh` that asserts every colon-separated entry of the verify.sh filter resolves to >= 1 test via `--gtest_list_tests`.
- Or move the filter into `CMakeLists.txt` next to the other two and have `verify.sh` read it, removing the hand-sync note at `CMakeLists.txt:926-927`.

## Acceptance criteria
- [ ] A test in the `unit` label fails when any one of the 12 filter entries is renamed, verified by mutation: rename `ForwardPassTest` and confirm the guard goes red.
- [ ] The guard reports the per-entry match count, so an entry resolving to 0 is named rather than inferred.

## Unverified
- The silent-pass behaviour for a non-matching plain `TEST` suite is asserted from `tests/CLAUDE.md:43-45`, which documents it for `TEST_P`; it was not re-demonstrated for a plain `TEST` suite. The Repro command that would settle it is marked NOT RUN.

**Comment by kekzl** [comment]
Fixed in d979e94a (#1700). Verified on main.
